src/hdmf/container.py

```diff
@@ -830,3 +833,218 @@ def __build_conf_methods(cls, conf_dict, conf_index, multi):
         get = conf_dict.get('get')
         if get is not None:
             setattr(cls, get, cls.__make_get(get, attr, container_type))
+
+
+class Table(Data):
+    r'''
```

Comment:
I assume this for representing compound tables, correct? If so, I think it would be be useful to mention in the docstring what the difference between Table and DynamicTable is (i.e., one is a column-store allows you to add columns, whereas, Table is a row-store that allows us to store a table in a single data array)

Reply:
Yeah, this is for working with datasets that have compound dtypes. It was formerly `NWBTable` in `pynwb`